### PR TITLE
Add UBSan sample

### PR DESCRIPTION
**Pull request by voltur01**

### CODE REVIEW

File: samples/src/cpp-baremetal-semihosting-ubsan/libubsan.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    HANDLER(invalid_objc_cast, "invalid-objc-cast")
+    HANDLER(function_type_mismatch, "function-type-mismatch")
+    HANDLER(implicit_conversion, "implicit-conversion")
+    HANDLER(nonnull_arg, "nonnull-arg")
+    HANDLER(nonnull_return, "nonnull-return")
+    HANDLER(nullability_arg, "nullability-arg")
+    HANDLER(nullability_return, "nullability-return")
+    HANDLER(pointer_overflow, "pointer-overflow")
+    HANDLER(cfi_check_fail, "cfi-check-fail")
+}
```
Comment (mplatings) [inline samples/src/cpp-baremetal-semihosting-ubsan/libubsan.cpp:54]: terminating newline

File: samples/src/cpp-baremetal-semihosting-ubsan/libubsan.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
+extern "C" {
+    #include <stdio.h>
```
Comment (mplatings) [inline samples/src/cpp-baremetal-semihosting-ubsan/libubsan.cpp:2]: Having includes in an `extern "C"` block seems like something to avoid.

**Review by mplatings — CHANGES_REQUESTED** [review]
Can this be added to the list of samples in CMakeLists.txt, to ensure it doesn't bitrot?

**Comment by voltur01** [comment]
Thank you for the review: I fixed the year and added to the list of samples as in the other sample; also updated libubsan.cpp as per your suggestions.

**Review by mplatings — APPROVED** [review]

**Pull request merged**